frontend/app/controllers/subjects_controller.rb

```diff
@@ -52,6 +61,36 @@ def update
                 })
   end
 
+  def defaults
```

Comment:
There are some slight variations, but yes it probably could be factored out. I didn't see any examples of mixin behavior in the controllers so far. Maybe at some point I can investigate this or maybe this would be a good reason to move to Rails 4 and use concerns.